### ISSUE TITLE
An issue with jumping cursor

### ISSUE THREAD

**Issue by dmpe** [editorconfig/editorconfig-vscode#268]
- [x] I tried running `code --disable-extensions` and the issue did **NOT** present itself.

## Issue

|             | Visual Studio Code | editorconfig-vscode |
|-------------|--------------------|---------------------|
| **Version** |  1.44         |  latest available on the marketplace   0.14.5    |

### Root `.editorconfig` File

```ini
# S.a. https://editorconfig.org/

root = true

[*]
indent_style = space
end_of_line = lf
charset = utf-8
indent_size = 2
trim_trailing_whitespace = true
insert_final_newline = true

# 4 space indentation
[*.py]
indent_style = space
indent_size = 4
```

Are there any other relevant `.editorconfig` files in your project? Yes / No

| Visual Studio Code Setting     | Default | User    | Workspace |
|--------------------------------|---------|---------|-----------|
| `editor.insertSpaces`          | `true`  | `____`  | `____`    |
| `editor.tabSize`               | `4`     | `_`     | `_`       |
| `editor.trimAutoWhitespace`    | `true`  | `____`  | `____`    |
| `files.autoSave`               | `"off"` | on | on  |
| `files.insertFinalNewline`     | `false` | `_____` | `_____`   |
| `files.trimTrailingWhitespace` | `false` | `_____` | `_____`   |

### Additional comments or steps to reproduce

I am facing an issue that when I programming on Python file, cursor starts to jump backwards.
Particularly when writing very fast, the only solution for me is to turn it off. 

For example:

"abcdgtgh|"

where | is current location of the cursor

all of the sudden jumps to 

"ab|cdgtgh"

Is anybody else seeing this behavior with Python. I am very confident that this has something to do with this plugin.

**Comment by seanwash** [comment]
Im seeing this as well with PHP. I disabled all plugins except for this one and this issues only appears when I have this plugin enabled.

**Comment by codejnki** [comment]
I've been having the same problem.  Try disabling the VSCode setting `trim_auto_whitespace`.  I've disabled that and so far have not seen my cursor jump around.

**Comment by codejnki** [comment]
While not ideal, I appear to be haveing some success by setting 

`trim_trailing_whitespace = false` at the `[*]` level.

I've then set the workspace setting `"files.trimTrailingWhitespace": true`

**Comment by codejnki** [comment]
@seanwash or @dmpe what is your `editor.renderWhitespace` set to?

**Comment by dmpe** [comment]
@codejnki mine is always set to "selection" (vscode default ?) and I did not even know it existed until today. :)

**Comment by codejnki** [comment]
Try this:

VSCode Settings
`"editor.renderWhitespace" : "none"`
`"files.trimTrailingWhitespace" : true`

.editorconfig
```
[*]
trim_trailing_whitespace = false
```
I'm curious if you see the same results I do.

**Comment by codejnki** [comment]
I'm sad to report that this has not solved my cursor jumping problem.  I'm still trying to see if I can find the magic settings combination to make this not frustrating.

**Comment by bijancamp** [comment]
Removing the `end_of_line` setting has fixed the issue for me in Babel JavaScript files on macOS:

```
# editorconfig.org
root = true

[*]
charset = utf-8

# end_of_line = lf

indent_size = 2
indent_style = space
insert_final_newline = true
trim_trailing_whitespace = true

[*.md]
trim_trailing_whitespace = false
```

**Comment by anavarre** [comment]
Could someone please edit the issue title to drop Python from the title? It's very much happening with PHP/Drupal.

Drupal's `.editorconfig` file here: https://git.drupalcode.org/project/drupal/-/blob/9.0.x/.editorconfig#L13

We're seeing the issue with Theia when autosave is on and saves automatically after 1 second, which makes things unusable.

**Comment by jedmao** [comment]
There was [an issue with this extension performing `TextEdit` operations when it shouldn't have](https://github.com/editorconfig/editorconfig-vscode/issues/288). After #292, this should only happens if there are actual changes that the extension needs to make, as it [saves your current selections](https://github.com/editorconfig/editorconfig-vscode/blob/a505935fbbd685eaea386f0d9acb6139a3406e30/src/DocumentWatcher.ts#L86), [applies the edits](https://github.com/editorconfig/editorconfig-vscode/blob/a505935fbbd685eaea386f0d9acb6139a3406e30/src/DocumentWatcher.ts#L92-L94) and then [restores your original selections](https://github.com/editorconfig/editorconfig-vscode/blob/a505935fbbd685eaea386f0d9acb6139a3406e30/src/DocumentWatcher.ts#L96). This explains why the issue presents itself when you are typing fast with auto-save turned on.

Auto-saving every 1 second sounds extremely aggressive to me, but I think y'all would be more upset if your selections weren't preserved. I think I need to close this as a won't fix, unless someone has some brilliant ideas.
